mbof/views.py

```diff
@@ -66,9 +68,37 @@ class EventViewSet(OcellusViewSet):
     queryset = Event.objects.all().order_by('postingTime')
     serializer_class = EventSerializer
 
+    def retrieve(self, request, *args, **kwargs):
+        log_action(request)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        serializer_data = serializer.data
+        if serializer_data['owner'] == self.request.user.username:
+            serializer_data['owner'] = "True"
+        else:
+            serializer_data['owner'] = "False"
+        return Response(serializer_data)
+
     def perform_create(self, serializer):
         log_action(self.request)
-        serializer.save(owner=self.request.user.__str__())
+        serializer.save(owner=self.request.user.username)
+
+    def perform_update(self, serializer):
+        log_action(self.request)
+        queryset = Event.objects.filter(owner=self.request.user, id=self.kwargs['pk'])
+        if not queryset.exists():
+            raise PermissionDenied("You do not have permission to edit")
+        serializer.save()
+
+    def get_queryset(self):
+        log_action(self.request)
+        query_set = Event.objects.all().order_by('postingTime')
+        for event in query_set:
+            if event.owner == self.request.user.username:
+                event.owner = True
+            else:
+                event.owner = False
```

Comment:
This loop changes the owner attribute of each event into a Boolean indicating whether the user is the owner of each one, right?  Loops for this purpose are less efficient than functions.  Consider using `map()` instead.

I wonder:  Why is this done here?  Why not let the code that uses these event objects do the comparison instead?